### PR TITLE
Move deprecated module checker to DeprecatedMixin

### LINKED ISSUE #4086: Make module deprecation checks reusable to plugins
### Is your feature request related to a problem? Please describe
Imports checker contains checks for deprecated modules:

pylint/pylint/checkers/imports.py

https://github.com/PyCQA/pylint/blob/d825c21e18de5618cfcacea0fd818efa80a6b0fe/pylint/checkers/imports.py#L229-L233

Unfortunately, there is no easy way how it can be simply reused to add support for deprecation checks in pylint plugins.

### Describe the solution you'd like
Extend solution proposed in #4049  and #4051 to support also deprecated modules

### Additional context
The final plugin can look like as follows:
```python
import astroid

from pylint.checkers import BaseChecker, DeprecatedMixin, utils
from pylint.interfaces import IAstroidChecker


class DeprecationChecker(BaseChecker, DeprecatedMixin):
    __implements__ = (IAstroidChecker,)
    name = "deprecated"
    msgs = {
        "W0002": (
            "Using deprecated method %s()",
            "deprecated-method",
            "The method is marked as deprecated and will be removed in "
            "the future.",
        ),
        "W0003": (
            "Using deprecated module %s()",
            "deprecated-module",
            "The module is marked as deprecated and will be removed in "
            "the future.",
        ),

    }

    def visit_call(self, node):
        try:
            for inferred in node.func.infer():
                # Calling entry point for deprecation check logic.
                self.check_deprecated_method(node, inferred)
        except astroid.InferenceError:
            return

    def visit_import(self, node):
        """triggered when an import statement is seen"""
        names = [name for name, _ in node.names]
        for name in names:
            self._check_deprecated_module(node, name)

    def visit_importfrom(self, node):
        """triggered when a from statement is seen"""
        basename = node.modname
        self.check_deprecated_module(node, basename)

    def deprecated_methods(self):
        return {'mymodule.deprecated_function', 'mymodule.MyClass.deprecated_method'}

    def deprecated_modules(self):
        return {'deprecated_module1', 'deprecated_module2'}


def register(linter):
    linter.register_checker(DeprecationChecker(linter))
```

### PR DESCRIPTION
**Pull request by matusvalo**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
This PR logic of checking the use of deprecated modules to DeprecatedMixin class.
Hence, it can be easilly reused this way.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :hammer: Refactoring  |

## Related Issue

Closes #4086

### CODE REVIEW

File: pylint/checkers/deprecated.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ class DeprecatedMixin:
             "deprecated-argument",
             "The argument is marked as deprecated and will be removed in the future.",
         ),
+        "W0402": (
+            "Uses of a deprecated module %r",
+            "deprecated-module",
+            "Used a module marked as deprecated is imported.",
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/deprecated.py:38]:
```suggestion
            "A module marked as deprecated is imported.",
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for this MR ! Sorry for the long review time.

File: ChangeLog [hunk 2]
```diff
@@ -15,6 +15,9 @@ Release date: TBA
   alternative to ``extension-pkg-whitelist`` and the message ``blacklisted-name`` is now emitted as
   ``disallowed-name``. The previous names are accepted to maintain backward compatibility.
 
+* Move deprecated checker to DeprecatedMixin
+
+  Closes #4086
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]:
```suggestion
* Move deprecated checker to ``DeprecatedMixin``
  
  Closes #4086

```

**Comment by cdce8p** [comment]
@Pierre-Sassoulas You need to add an empty line after `Closes` in the changelog to make the pre-commit check happy.

File: ChangeLog [hunk 3]
```diff
@@ -15,6 +15,9 @@ Release date: TBA
   alternative to ``extension-pkg-whitelist`` and the message ``blacklisted-name`` is now emitted as
   ``disallowed-name``. The previous names are accepted to maintain backward compatibility.
 
+* Move deprecated checker to ``DeprecatedMixin``
+  
+  Closes #4086
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]:
```suggestion
  Closes #4086


```

**Comment by Pierre-Sassoulas** [comment]
I swear I need a github feature to apply pre-commit auto-fix on a pull request 😄

**Comment by cdce8p** [comment]
😄 Seems like there are some trailing whitespaces now

File: ChangeLog [hunk 4]
```diff
@@ -15,6 +15,10 @@ Release date: TBA
...
+
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]:
```suggestion
* Move deprecated checker to ``DeprecatedMixin``

  Closes #4086

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for your help @cdce8p a 5th edit would have been embarrassing, you know ? 😄

**Pull request merged**